### ISSUE TITLE
Extension of fragments does not work

### ISSUE THREAD

**Issue by asrichter** [fidelram/deepTools#140]
The recently changed `--extendReads` option does not work. There are several issues.

* When single-end data is provided and a value for the fragment size is specified **together with** `--normalizeTo1x`, bamCoverage crashes.
```
$ bamCoverage -b H3K4Me3.bam -o H3K4me3.bw --normalizeTo1x 2451960000 --extendReads 100
Traceback (most recent call last):
...
  File "lib/python2.7/site-packages/deeptools/bamCoverage.py", line 175, in main
    funcArgs = {'scaleFactor': get_scale_factor(args)}
  File "lib/python2.7/site-packages/deeptools/bamCoverage.py", line 136, in get_scale_factor
    fragment_length = args.fragmentLength
AttributeError: 'Namespace' object has no attribute 'fragmentLength'
```
* When single-end data is provided and **no** fragment size is specified **together with** `--normalizeTo1x`, bamCoverage crashes with another error.
```
$ bamCoverage -b H3K4Me3.bam -o H3K4me3.bw --normalizeTo1x 2451960000 --extendReads
Traceback (most recent call last):
...
  File "lib/python2.7/site-packages/deeptools/bamCoverage.py", line 175, in main
    funcArgs = {'scaleFactor': get_scale_factor(args)}
  File "lib/python2.7/site-packages/deeptools/bamCoverage.py", line 128, in get_scale_factor
    fragment_length = args.frag_len_dict['median']
AttributeError: 'Namespace' object has no attribute 'frag_len_dict'
```
* When single-end data is provided and **no** fragment size is specified **without** `--normalizeTo1x`, bamCoverage uses a default fragment length of 0.0 and crashes with a wrong error instead of complaining that no value was passed.
```
$ bamCoverage -b H3K4Me3.bam -o H3K4me3.bw --extendReads
...
defaultFragmentLength: 0.0
Traceback (most recent call last):
...
  File "lib/python2.7/site-packages/deeptools/bamCoverage.py", line 193, in main
    format=args.outFileFormat, smooth_length=args.smoothLength)
  File "lib/python2.7/site-packages/deeptools/writeBedGraph.py", line 135, in run
    numberOfProcessors=self.numberOfProcessors)
  File "lib/python2.7/site-packages/deeptools/mapReduce.py", line 135, in mapReduce
    res = pool.map_async(func, TASKS).get(9999999)
  File "/usr/lib64/python2.7/multiprocessing/pool.py", line 554, in get
    raise self._value
AssertionError: fragment start greater than fragmentend for read SRR037602.14704554
```

**Comment by fidelram** [comment]
Fixed in #149

**Comment by sotuamax** [comment]
I encountered the same issue today. 
`AssertionError: fragment start greater than fragmentend for `

Is there any example to show how the '-e' together with "--maxFragmentLength" and "--minFragmentLength" can be used? I also tried to sort the BAM using name, and apparently it didn't help solve the problem. 

Thank you. 

Best,
